### ISSUE TITLE
Storybook: Modal component needs to disable the background

### ISSUE THREAD

**Issue by marcysutton** [forem/forem#11726]
**Describe the bug**

Related to #10610, the Modal Component in Storybook needs to disable interactive elements in the background. I opened this issue separately so Storybook wouldn't drift from the live site since there are multiple kinds of modals. (But if this issue is truly a duplicate of #10610 implementation-wise, it's ok to close it)

**To Reproduce**

1. Go to https://storybook.forem.com/?path=/story/components-modals--default
2. Tab through the element canvas
3. Observe keyboard focus going behind the modal.

**Expected behavior**

The background should be made inert so that keyboard focus doesn't go behind the modal. It could be done with the `inert` attribute on sibling content, or by walking the DOM to add `tabindex="-1"` on every interactive control and `aria-hidden="true"` on sibling wrapper elements in the DOM subtree.

**Screenshots**

<img width="1358" alt="Focus console logging in the background behind a modal in Storybook" src="https://user-images.githubusercontent.com/1045233/100974369-01fc3280-34f1-11eb-8013-665255ad63e7.png">

**Desktop (please complete the following information):**

- OS, version: All
- Browser, version: All

**Smartphone (please complete the following information):**

- Device: All
- OS, version: All
- Browser, version: All

**Additional context**

This is part of a broader review of Storybook components, so additional issues will be filed separately.

**Comment by aitchiss** [comment]
Closed by #12511
